### PR TITLE
Datastore dump ui interface

### PR DESCRIPTION
**Pull request by tino097**
As it was mentioned in #3821, i've tried to add interface to be used for custom data dump formats from extensions.  

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckanext/datastore/controller.py [hunk 1]
```diff
@@ -15,24 +15,20 @@
...
 import ckan.lib.navl.dictization_functions as dict_fns
-
+from ckan.common import config
 from itertools import izip_longest
 
 int_validator = get_validator('int_validator')
 boolean_validator = get_validator('boolean_validator')
 
-DUMP_FORMATS = 'csv', 'tsv', 'json', 'xml'
+DUMP_FORMATS = config.get('ckan.dump_formats')
```
Comment (klikstermkd) [inline ckanext/datastore/controller.py]: @tino097 If `ckan.dump_formats` setting is not set in the `.ini` file, the `DUMP_FORMATS` variable will be `None`.

Comment (klikstermkd) [reply ckanext/datastore/controller.py]: Also, if it's set like this `ckan.dump_formats = csv tsv json xml` it should be parsed to be stored as an array.

File: ckanext/datastore/controller.py [hunk 2]
```diff
@@ -51,7 +47,7 @@ def dump(self, resource_id):
 
         if fmt not in DUMP_FORMATS:
             abort(400, _(
-                u'format: must be one of %s') % u', '.join(DUMP_FORMATS))
```
Comment (wardi) [inline ckanext/datastore/controller.py]: joining with `u''` will jam all the formats together in an unreadable way

File: ckanext/datastore/writer.py [hunk 3]
```diff
@@ -3,13 +3,39 @@
...
+        records_format = u'tsv'
+    elif fmt == u'json':
+        writer_factory = json_writer
+        records_format = u'lists'
+    elif fmt == u'xml':
+        writer_factory = xml_writer
+        records_format = u'objects'
+    else:
+        writer_factory = plugin_writer
+        records_format = u'objects'
```
Comment (wardi) [inline ckanext/datastore/writer.py]: why wouldn't a writer plugin want to be able to use one of the more efficient dump formats?

**Comment by kmbn** [comment]
We decided to close old pull requests that are not actively worked on so that we can focus our effort and attention on PRs made against the current version of CKAN. A lot has changed in the last several months and many PRs will have gone stale or may now be in conflict with the latest version.

Please note: this isn't a judgement about the quality of your pull request. We just don't have the resources to review older PRs.

If you would like to reopen your PR (and update it if necessary), please do!

**Pull request closed without merging**